logcheck/pkg/logcheck.go

```diff
@@ -397,6 +397,13 @@ func isKeysValid(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funNam
 				Message: fmt.Sprintf("Key positional arguments %s are expected to be lowerCamelCase alphanumeric strings. Please remove any non-Latin characters.", lit.Value),
 			})
 		}
+		filterChar := []string{` `, `[`, `]`, `%`, `$`, `*`}
```

Comment:
Instead of blacklisting certain characters, let's do a full check for compliance with https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments:
- lowerCamelCase
- alphanumeric

Lower camel case is tricky: acronyms are a valid (undocumented...) exception. `CIDR` was used as example.

I think we must allow `^[A-Z]{2+}[[:alnum:]]*` as an alternative to `^[a-z][[:alnum:]]*$`.

Probably the best approach is to match the string against a regular expression and warn about all keys that don't match the expression.

This is a new, stricter check than before. Let's add a separate enable configuration for it ("key-check"?) because not everyone might want to follow the Kubernetes recommendations.

Reply:
sounds good. let me try to complete